pkg/builder/local_build_executor.go

```diff
@@ -356,6 +382,13 @@ func (be *localBuildExecutor) Execute(ctx context.Context, filePool re_filesyste
 		response.Result.StderrDigest = stderrDigest.GetPartialDigest()
 	}
 
+	outputParentDirectories, err := newOutputDirectories(command, inputRootDirectory, false)
+	if err != nil {
+		attachErrorToExecuteResponse(response, err)
+		return response
+	}
+	defer outputParentDirectories.Close()
+
 	// Upload output files.
 	for _, outputFile := range command.OutputFiles {
 		outputParentDirectory := outputParentDirectories[path.Dir(outputFile)]
```

Comment:
If we're going down this road, we should likely partially revert 855598d6fec55bfa0d15f5d1d58d55a061b04b4e. It may now well be the case that output directories no longer exist or have become symlinks. Those cases now also need to be handled.

Reply:
I think I have now untangled this.